girder/api/describe.py

```diff
@@ -135,7 +136,15 @@ def asDict(self):
             resp['consumes'] = self._consumes
 
         if self._produces:
-            resp['produces'] = self._produces
+            # swagger has a bug where not all appropriate mime types are
+            # considered to be binary (see
+            # https://github.com/swagger-api/swagger-ui/issues/1605).  If we
+            # have specified zip format, replace it with
+            # application/octet-stream
+            produces = list(OrderedDict.fromkeys([
+                'application/octet-stream' if item in ('application/zip', )
+                else item for item in self._produces]))
```

Comment:
Why `list(OrderedDict())`?

Reply:
I wanted to remove duplicates from the list but not change its order.  This is the most compact pythonic way I know to do that.

Reply:
Can you add a comment that expresses that intention? Otherwise someone might come along and not realize you are trying to preserve the order and change it to use `set()` or something.